build2.py

```diff
@@ -142,6 +150,7 @@ def construct_graph(directory):
     # get all immediate subdirectories
     recipe_dirs = next(os.walk(directory))[1]
     recipe_dirs = set(x for x in recipe_dirs if not x.startswith('.'))
+    changed_recipes = git_changed_files('HEAD')
```

Comment:
Need a better way of getting git revision (preferably a command line argument).  I'll change this tomorrow.